src/node_api_jsrt.cc

```diff
@@ -241,23 +241,49 @@ class ExternalCallback {
   void* _data;
 };
 
+inline napi_status JsCopyUtf8Bytes(JsValueRef key,
+                                   char** buffer, size_t* copiedBytesCount) {
+  int strLength = 0;
+  CHECK_JSRT(JsGetStringLength(key, &strLength), napi_string_expected);
+
+  // assume string contains ascii characters only
+  *buffer = reinterpret_cast<char*>(malloc(strLength + 1));
```

Comment:
Actually, the function that calls `JsCopyUtf8Bytes` is going to leak the allocation in the new scheme, since the old code used the value locally and never passed it back out, while the new code will just leak.